src/js/popup/popup.js

```diff
@@ -1025,77 +1030,66 @@
-            // AND at least one checkbox is checked
-            if (popup.utilities.isSortaAURL(inputFieldOtherDetails.value)) {
-              reportIssueSubmitBtn.disabled = false;
-            }
+            popup.panel.webcompat.validateForm(formData);
           });
 
           inputFieldOtherDetails.placeholder = browser.i18n.getMessage(
             "popupReportIssueCaseOtherDetails"
           );
         },
-        handleReportIssueFormSubmission: async (event) => {          
-          event.preventDefault();
-          const data = new FormData(event.target);
-          const reportData = Object.fromEntries(data.entries());
-          reportData.user_agent = await getBrowser();
+        validateForm: (formData) => {
+          // Check if inputFieldUrl is valid
+          const inputFieldUrlIsNotEmpty = formData.inputFieldUrl.checkValidity();
 
-          Object.keys(reportData).forEach(function (value) {
-            // Switch "on" to true
-            if (reportData[value] === "on") {
-              reportData[value] = true;
-            }
-            // Remove from report if empty string
-            if (reportData[value] === "") {
-              delete reportData[value];
-            }
-          });
+          // Ensure that the custom input looks like a URL without https:// or http:// (e.g. test.com, www.test.com) 
+          const inputFieldUrlIsSortaAUrl = popup.utilities.isSortaAURL(formData.inputFieldUrl.value)
 
-          // Clean URL data to add "http://" before it if the custom input doesn't contain a HTTP protocol
+          // Validate that at least one checkbox is checked
+          const checkboxes = formData.checkboxes
+          const isACheckBoxChecked = [...checkboxes].some(e => e.checked == true);
+
+          // If the "other" checkbox is checked, confirm the "other" input is valid
+          const checkedCheckboxes = [...checkboxes].filter(e => e.checked == true);          
+          const isCheckedCheckboxOther = checkedCheckboxes.some(e => e.id == "issue-case-other");
+          const inputFieldOtherDetailsIsValid = formData.inputFieldOtherDetails.checkValidity();
+
+          // This tests for two possible valid form states: 
+          // A: User has URL field filled out correctly AND at least one reason checkbox checked (Not OTHER checkbox)
+          // B: User has URL field filled out correctly AND has checked the OTHER checbox AND filled out the other input form correctly
           if (
-            !(
-              reportData.issue_on_domain.startsWith("http://") ||
-              reportData.issue_on_domain.startsWith("https://")
-            )
+              (inputFieldUrlIsNotEmpty && inputFieldUrlIsSortaAUrl && isACheckBoxChecked && !isCheckedCheckboxOther) || 
+              (inputFieldUrlIsNotEmpty && inputFieldUrlIsSortaAUrl && isACheckBoxChecked && isCheckedCheckboxOther && inputFieldOtherDetailsIsValid) 
```

Comment:
I'm not happy with this SUPER long chained IF. _(Two actually??)_ 

@lloan Looking at you for an enhancement. 😅

Reply:
We could maybe play around with something like this? 

```validateForm: (formData) => {
  // Validate inputFieldUrl
  const inputFieldUrlIsValid = formData.inputFieldUrl.checkValidity() && popup.utilities.isSortaAURL(formData.inputFieldUrl.value);

  // Validate checkboxes
  const checkboxes = formData.checkboxes;
  const isACheckBoxChecked = checkboxes.some(e => e.checked === true);
  const isCheckedCheckboxOther = checkboxes.some(e => e.checked === true && e.id === "issue-case-other");
  const inputFieldOtherDetailsIsValid = formData.inputFieldOtherDetails.checkValidity();


  // Determine the form's validity
  const formIsValid = inputFieldUrlIsValid && isACheckBoxChecked && (!isCheckedCheckboxOther || (isCheckedCheckboxOther && inputFieldOtherDetailsIsValid));

  // Set the form's state
  formData.reportIssueSubmitBtn.disabled = !formIsValid;
  formData.form.dataset.formIsValid = formIsValid;
},

Reply:
Could probably be a TODO tbh.

Reply:
Ahhh!! `&&` on a `const` 🥲 

It's beautiful. Thanks for the solve.

Reply:
Revised the two `inputField...` buttons to `inputFieldUrlIsValid` in f307c658f813e6b9414cc92425135c52b9647152. Left the rest of the logic as is for now.